src/ua_securechannel.h

```diff
@@ -105,6 +105,11 @@ struct UA_SecureChannel {
     UA_ConnectionManager *connectionManager;
     uintptr_t connectionId;
 
+    /* Reverse connect */
+    uintptr_t reverseConnectListenConnectionId;
```

Comment:
You don't any of these three.
Simply reuse the `connectionId` also for the listen socket.
And replace with the "active socket" once it opens.

The boolean flags are not needed either.
The `UA_SECURECHANNELSTATE_REVERSE_LISTENING` and `UA_SECURECHANNELSTATE_REVERSE_CONNECTED` flags ought to be enough.